src/preferencesControl.ts

```diff
@@ -164,12 +164,49 @@ export class PreferencesControl {
      * 4. Click "Reset all" button, cookieCategoriesPreferences will be reset
      */
     private addPreferencesButtonsEvent(): void {
-        let closeModalIcon = document.getElementsByClassName(styles.closeModalIcon)[0];
+        let closeModalIcon: HTMLElement = <HTMLElement> document.getElementsByClassName(styles.closeModalIcon)[0];
 
         let cookieItemRadioBtn: Element[] = [].slice.call(document.getElementsByClassName(styles.cookieItemRadioBtn));
-        let modalButtonSave: HTMLInputElement = <HTMLInputElement>document.getElementsByClassName(styles.modalButtonSave)[0];
+        let modalButtonSave: HTMLInputElement = <HTMLInputElement> document.getElementsByClassName(styles.modalButtonSave)[0];
         let modalButtonReset: HTMLInputElement = <HTMLInputElement> document.getElementsByClassName(styles.modalButtonReset)[0];
 
+        let lastAcceptRadioBtn: HTMLElement | null = null;
+        let lastRejectRadioBtn: HTMLElement | null = null;
+        if (cookieItemRadioBtn && cookieItemRadioBtn.length) {
+            lastAcceptRadioBtn = <HTMLElement> cookieItemRadioBtn[cookieItemRadioBtn.length - 1];
+            lastRejectRadioBtn = <HTMLElement> cookieItemRadioBtn[cookieItemRadioBtn.length - 2];
+        }
+
+        let lastElementTab = function(event: KeyboardEvent): void {
+            if (event.keyCode == 9 && !event.shiftKey) {
+                event.preventDefault();
+                closeModalIcon.focus();
+            }
+        };
+        let firstElementShiftTab = function(event: KeyboardEvent): void {
+            if (event.keyCode == 9 && event.shiftKey) {
+                event.preventDefault();
+                lastAcceptRadioBtn?.focus();
+            }
+        };
+
+        if (modalButtonReset && modalButtonSave && modalButtonReset.disabled && modalButtonSave.disabled) {
```

Comment:
This logic on checking if disabled wouldn't work correctly. You're checking only during the dialog creation buttons are dynamically enabled/disabled when user clicks on the radio buttons